great_tables/_options.py

```diff
@@ -969,3 +969,673 @@ def opt_all_caps(
+    borders applied, some apply darker colors to the table stub and summary sections, and, some even
+    have vertical lines. In addition to choosing a `style` preset, there are six `color` variations
+    that each use a range of five color tints. Each of the color tints have been fine-tuned to
+    maximize the contrast between text and its background. There are 36 combinations of `style` and
+    `color` to choose from.
+
+    Parameters
+    ----------
+    style : int
+        Six numbered styles are available. Simply provide a number from `1` (the default) to `6` to
+        choose a distinct look.
+    color : str
+        The color scheme of the table. The default value is `"blue"`. The valid values are `"blue"`,
+        `"cyan"`, `"pink"`, `"green"`, `"red"`, and `"gray"`.
+
+    Returns
+    -------
+    GT
+        The GT object is returned. This is the same object that the method is called on so that we
+        can facilitate method chaining.
+
+    Examples
+    --------
+    Using select columns from the `exibble` dataset, let's create a table with a number of
+    components added. Following that, we'll apply a predefined style to the table using the
+    `opt_stylize()` method.
+
+    ```{python}
+    from great_tables import GT, exibble, md
+
+    gt_tbl = (
+          GT(
+            exibble[["num", "char", "currency", "row", "group"]],
+            rowname_col="row",
+            groupname_col="group"
+          )
+          .tab_header(
+            title=md("Data listing from **exibble**"),
+            subtitle=md("`exibble` is a **Great Tables** dataset.")
+          )
+          .fmt_number(columns="num")
+          .fmt_currency(columns="currency")
+          .tab_source_note(source_note="This is only a subset of the dataset.")
+          .opt_stylize()
+        )
+
+    gt_tbl
+    ```
+
+    The table has been stylized with the default style and color. The default style is `1` and the
+    default color is `"blue"`. The default style is a combination of color and border settings that
+    are applied to the table. The default style and color can be modified by providing the `style=`
+    and `color=` arguments.
+
+    ```{python}
+    gt_tbl.opt_stylize(style=2, color="green")
+    ```
+    """
+
+    # Validate the `style` and `color` arguments
+    if style not in [1, 2, 3, 4, 5, 6]:
+        raise ValueError("`style` must be an integer value from `1` to `6`.")
+    color = _utils._match_arg(x=color, lst=["gray", "blue", "cyan", "pink", "green", "red"])
+
+    # Get the style parameters based on the `style` and `color` arguments
+    params = _dict_styles_colors_params[f"{color}-{style}"]
+
+    # Apply the style parameters to the table using the `tab_options()` method
+    # TODO: the commented out lines are for future use when:
+    #   (1) row striping is implemented
+    #   (2) summary rows are implemented
+    #   (3) grand summary rows are implemented
+    res = tab_options(
```

Comment:
I wonder if it would be useful to create a mapper, that takes the style parameters in its init, and can output the tab_options values as a dict.

e.g.

```
from dataclasses import dataclass, fields
from typing import ClassVar

@dataclass
class StyleMapper:
    table_hlines_color: str
    column_labels_background_color: str

    mappings: ClassVar[dict[str, list[str]]] = {
        "table_hlines_color": ["table_border_top_color", "table_border_bottom_color"],
        "column_labels_background_color": ["column_labels_background_color"]
    }

    def map_entry(self, name: str):
        return {k: getattr(self, name) for k in self.mappings[name]}

    def map_all(self):
        items = {}
        for field in fields(self):
            items.update(self.map_entry(field.name) )
        return items

StyleMapper("ABC", "DEF").map_all()
```

Then you could pass the result straight to tab_options, e.g.

```python
# inside opt_stylize
self.tab_options(**StyleMapper(...).map_all())
```

What is helpful about this for me is that it signals that:

* StyleMapper is a very simple one entry thing to multiple entries in a dict
* opt_stylize is a thin wrapper around tab_options

Reply:
Feel free to punt this to an issue and just link here

Reply:
Was able to integrate this nicely. Thanks so much for working through this with me!